### ISSUE TITLE
remove reasonability checks

### ISSUE THREAD

**Issue by malteish** [corpus-io/tokenize.it-smart-contracts#247]
In `Vesting.sol`, we do these checks:
```
require(
            _start >= block.timestamp - TIME_HORIZON && _start <= block.timestamp + TIME_HORIZON,
            "Start must be reasonable"
        );
require(_cliff >= 0 && _cliff <= _duration, "Cliff must be reasonable");
require(_duration > 0 && _duration <= TIME_HORIZON, "Duration must be reasonable");
```

We can not perform the same checks when accepting a commitment. We used to check when revealing a commitment. Since revealing might happen a long time for now, the checks might fail then (e.g. because the start date was 18 years ago when the commitment was created, but 4 years later, when a reveal is attempted, it is 22 years ago, so the reveal fails). 
I moved the checks to `createVesting` from `_createVesting`. So now they only apply to transparent vesting plan creation, which solves that issue. But this is incomplete - we check one way of creating a plan, but not the other. I propose removing the 3 checks above completely.

**Comment by CJentzsch** [comment]
The check for the cliff always makes sense, as it is just relative to the duration.
For the other two, I would say, let'e keep them in `_createVesting` and change TIME_HORIZON to 100 years.

**Comment by malteish** [comment]
I know this is not a really important topic, but. In 200 years from now, Gloria, grand-daughter of, Valentina, an early supporter of meta-brew, who's slogan is "We lead the decentralized revolution of the brewing industry in the Web3 and the Metaverse and stay for the next 500 years.", wants to reveal their vesting. And she can't, because this check reverts. Meanwhile Anton, grand-grand son of Emilia, who was also an early supporter of meta-brew, lives a life of luxury because Emilia revealed her vesting plan immediately. Emilia was offered the same conditions as Valentina back in the day. Meta-Brew has seen 35 CEOs since then an nobody has access to the wallets that govern all of that legacy stuff anymore. Also, the company is close to bankrupt, so while suing is an option, the settlement could never compensate for Glorias inaccessible tokens.
Long story short: I want to keep conditions for blind vestings and public vestings equal. Since we can't to the check properly for blind vestings, I want to remove if for both.

**Comment by malteish** [comment]
summarizing a discussion with Christoph:
- removing the checks except for cliff < duration is okay. These checks have to take place off-chain.
- cliff > duration is a bad situation though: which of the two is valid? To keep the interests of the company issuing the tokens protected, but prevent a beneficiary from simply not being able to reveal, we decide to do `duration = max(cliff, duration);`. 
An employee will (or at least should) check their cliff straight away. And if the cliff is "one year and one second" and is displayed as "one year" in the frontend, indistinguishable from the duration, that really is exactly one year long, then this approach is a good compromise.

**Comment by malteish** [comment]
done https://github.com/corpus-io/tokenize.it-smart-contracts/pull/253
